internal/agent/event/event.go

```diff
@@ -7,7 +7,17 @@ import "context"
 // Name is a unique name identifying an event.
 type Name string
 
-// Event is a recordable event.
+// Event is a recordable event. Each event in the event package implements this interface. Consumers
```

Comment:
This is great, thanks. A few suggestions for things to add that I think would be helpful for implementors. 
- what "recordable event" is, means, and when/where events are recorded (i believe its just in the agent logic).
- transports implementations are responsible for sending the event. 
- concrete event types will contain more context that can be used when sending an event.

Reply:
I can add something for the first 2. Isn't the third point covered by the example?

Reply:
I've tweaked this a little but I decided not to make any mention of transport because thats 1 of possibly more consumers. I think the documentation in this package needs to be tight and relevant to the package itself.